### PR TITLE
Update initial_value parameters of generic_system tests

### PR DESCRIPTION
**Pull request by kumar-sanjeeev**
- Add initial values for every state interface in the ros2_control.xacro.
- Update `print_hint` to `false` to suppress warnings when no initial value is provided.

### CODE REVIEW

File: hardware_interface/src/mock_components/generic_system.cpp [hunk 1]
```diff
@@ -723,11 +723,11 @@ void GenericSystem::initialize_storage_vectors(
           if (it2 != component.parameters.end())
           {
             states[index][i] = hardware_interface::stod(it2->second);
-            print_hint = true;
+            print_hint = false;
```
Comment (christophfroehlich) [inline hardware_interface/src/mock_components/generic_system.cpp:726]:
```suggestion
            print_hint = true;
```

Comment (christophfroehlich) [reply hardware_interface/src/mock_components/generic_system.cpp:726]: we want to print the hint if the old parameter is used

File: hardware_interface/src/mock_components/generic_system.cpp [hunk 2]
```diff
@@ -723,11 +723,11 @@ void GenericSystem::initialize_storage_vectors(
...
           }
           else
           {
-            print_hint = true;
+            print_hint = false;
           }
```
Comment (christophfroehlich) [inline hardware_interface/src/mock_components/generic_system.cpp:731]: just delete the else branch -> we don't want to print a warning if no initial value is given

**Review by christophfroehlich — CHANGES_REQUESTED** [review]
Thanks for the PR! I changed the base branch to humble, it seems that you have forgotten to change the default.
When you commit another time, the checks will be triggered hopefully.

File: hardware_interface/src/mock_components/generic_system.cpp (hunk 1 above)
Comment (kumar-sanjeeev) [reply hardware_interface/src/mock_components/generic_system.cpp:726]: understood.

**Review by christophfroehlich — APPROVED** [review]
LGTM, there are no more deprecation warnings in the logs!

I don't think that we would need 
```
      <state_interface name="velocity">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="acceleration">
        <param name="initial_value">0.0</param>
      </state_interface>
```
everywhere, at least it isn't done in #1150 
This will only make a difference if `calculate_dynamics` is true

**Comment by kumar-sanjeeev** [comment]
> LGTM, there are no more deprecation warnings in the logs!
> 
> I don't think that we would need
> 
> ```
>       <state_interface name="velocity">
>         <param name="initial_value">0.0</param>
>       </state_interface>
>       <state_interface name="acceleration">
>         <param name="initial_value">0.0</param>
>       </state_interface>
> ```
> 
> everywhere, at least it isn't done in #1150 This will only make a difference if `calculate_dynamics` is true

Okay to be consistence with #1150, I will also remove them from here.

**Review by christophfroehlich — APPROVED** [review]

**Pull request merged**